src/Membership/MembershipRequestHandler.php

```diff
@@ -43,17 +43,26 @@ public function __construct(
     public function handle(MembershipRequest $membershipRequest)
     {
         $adherent = $this->adherentFactory->createFromMembershipRequest($membershipRequest);
-        $token = AdherentActivationToken::generate($adherent);
 
         $this->manager->persist($adherent);
+        $this->manager->flush();
+
+        $this->dispatcher->dispatch(UserEvents::USER_CREATED, new UserEvent($adherent));
+    }
+
+    public function join(Adherent $adherent, MembershipRequest $membershipRequest)
+    {
+        $adherent->updateMembership($membershipRequest, $this->addressFactory->createFromAddress($membershipRequest->getAddress()));
+        $token = AdherentActivationToken::generate($adherent);
+
         $this->manager->persist($token);
         $this->manager->flush();
 
         $activationUrl = $this->generateMembershipActivationUrl($adherent, $token);
         $this->mailer->sendMessage(AdherentAccountActivationMessage::createFromAdherent($adherent, $activationUrl));
```

Comment:
I think this message should be in `handle`

Reply:
The naming has lost all its sense, `membership`, `AdherentAccountActivationMessage`,  ...

Reply:
Agree